File: composite-pattern/java/src/OrganizationComponent.java
package src;

// 定义部件接口或抽象类，分支和叶子节点遵循该类约定
public interface OrganizationComponent {
   public void add(OrganizationComponent component);

   public void remove(OrganizationComponent component);

   public OrganizationComponent getChild(int index);

   public void operation();

   public String getName();
}

File: composite-pattern/go/src/OrganizationComponent.go
package src

// 定义部件接口或抽象类，分支和叶子节点遵循该类约定
type OrganizationComponent interface {
  Add(component OrganizationComponent)

  Remove(component OrganizationComponent)

  GetChild(index int) OrganizationComponent

  Operation()

  GetName() string
}
